src/main/java/net/mcreator/blockly/BlocklyToCode.java

```diff
@@ -124,10 +126,27 @@ public boolean checkIfStatementInputsProvide(Dependency dependency) {
 		return false;
 	}
 
-	public final void processBlockProcedure(List<Element> blocks) throws TemplateGeneratorException {
+	public final void processBlockProcedure(List<Element> blocks, StatementInput statementInput) throws TemplateGeneratorException {

```

Comment:
Do not add an additional argument, instead, use statementInputStack, and iterate through this stack to see if any of the 'StatementInput' in the stack specifies to disable local variables.